plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -2355,6 +2359,107 @@ public void testUnsupportedWriterVersion()
+
+        assertQuery("SELECT * FROM " + sourceTable, sourceTableValues);
+
+        String clonedTable = "test_clone_merge_cloned_" + randomNameSuffix();
+        Path cloneTableLocation = sourceLocation.resolveSibling(clonedTable);
+        copyDirectoryContents(new File(Resources.getResource(clonedResourceName).toURI()).toPath(), cloneTableLocation);
+
+        // Replace the fixed s3 prefix with actual(local) absolute path prefix
+        updateClonedTableDeletionVectorPathPrefix(cloneTableLocation, sourceLocation.toString());
+
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(clonedTable, cloneTableLocation.toUri()));
+
+        // cloned table without any changes
+        assertQuery("SELECT * FROM " + clonedTable, sourceTableValues);
+
+        // update on cloned table
+        @Language("SQL") String expectedValuesAfterUpdate = """
+                VALUES
+                (1, 'A', TIMESTAMP '2024-01-01'),
+                (2, 'updated', TIMESTAMP '2024-01-01'),
+                (3, 'C', TIMESTAMP '2024-02-02'),
+                (4, 'updated', TIMESTAMP '2024-02-02')
+        """;
+        assertUpdate("UPDATE " + clonedTable + " SET v = 'updated' WHERE id IN (2, 4)", 2);
+        assertQuery("SELECT * FROM " + clonedTable, expectedValuesAfterUpdate);
+
+        // merge on cloned table, including insert,update,delete
+        String mergeSql = """
+                MERGE INTO %s t
+                USING (VALUES (1, 'yyy', TIMESTAMP '2025-01-01'), (2, 'merged', TIMESTAMP '2025-02-02'), (5, 'kkk', TIMESTAMP '2025-03-03')) AS s(id, v, part)
+                ON (t.id = s.id)
+                WHEN MATCHED AND s.v = 'yyy' THEN DELETE
+                WHEN MATCHED THEN UPDATE SET v = s.v
+                WHEN NOT MATCHED THEN INSERT (id, v, part) VALUES(s.id, s.v, s.part)
+                """.formatted(clonedTable);
+
+        @Language("SQL") String expectedValuesAfterMerge = """
+                VALUES
+                (2, 'merged', TIMESTAMP '2024-01-01'),
+                (3, 'C', TIMESTAMP '2024-02-02'),
+                (4, 'updated', TIMESTAMP '2024-02-02'),
+                (5, 'kkk', TIMESTAMP '2025-03-03')
+                """;
+
+        assertUpdate(mergeSql, 3);
+        assertQuery("SELECT * FROM " + clonedTable, expectedValuesAfterMerge);
+
+        // source table not change
+        assertQuery("SELECT * FROM " + sourceTable, sourceTableValues);
+    }
+
+    public static void updateClonedTableDeletionVectorPathPrefix(Path location, String newPrefix)
+            throws IOException
+    {
+        String oldPrefix = "s3://test-bucket/tiny/clone_merge_deletion_vector_source";
+        Pattern pattern = Pattern.compile("(?<=\\\"pathOrInlineDv\\\":\\\")" + Pattern.quote(oldPrefix));
+
+        try (Stream<Path> stream = Files.walk(location)) {
+            stream.filter(file -> !Files.isDirectory(file))
+                    .filter(file -> file.getFileName().toString().endsWith(".json"))
```

Comment:
This will not update snapshot parquet file. Is it ok that it has incorrect paths? Maybe snapshot is not needed?

Reply:
Removed